int64 X_init
int64 Y_init
int64 Z_init